File: 1072-next-greater-node-in-linked-list/next-greater-node-in-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int[] nextLargerNodes(ListNode head) {
        ListNode temp=head;
        List<Integer> list=new ArrayList<>();
        while(temp!=null){
            list.add(temp.val);
            temp=temp.next;
        }
        int arr[]=new int[list.size()];
        for(int i=0;i<list.size();i++){
            for(int j=i+1;j<list.size();j++){
                if(list.get(i)<list.get(j)){
                    arr[i]=list.get(j);
                    break;
                }
            }
        }
        return arr;
    }
}

File: 1072-next-greater-node-in-linked-list/next-greater-node-in-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number[]}
 */
var nextLargerNodes = function(head) {
    let lst=[];
    while(head!==null){
        lst.push(head.val);
        head=head.next;
    }
    let arr=new Array(lst.length).fill(0);
    for(let i=0;i<lst.length;i++){
        for(let j=i+1;j<lst.length;j++){
            if(lst[i]<lst[j]){
                arr[i]=lst[j];
                break;
            }
        }
    }
    return arr;
};